### DISCUSSION TITLE
Examples for efficiently extending a MongoDB list field with multiple new items

### DISCUSSION THREAD

**Discussion by geyang** [prisma/prisma#13582] [Q&A]
In Mongo I can write a query that looks like the following:

```javascript
db.students.update(
   { name: "joe" },
   { $push: { scores: { $each: [ 90, 92, 85 ] } } }
)
```

and this will give me the following object:

```javascript
students: {
   name: "joe",
   scores: [100, 99,  90, 92, 85 ]
}
```

## What I Have Tried

First of all, iterating through the list items item-by-item is infeasible for our application. The example here is simple, but the list items in our application is larger.

So I tried to use `$

```javascript
  let rawQuery = {
          insert: "Student",
          bypassDocumentValidation: false,
          documents: [
            text
              ? {text, deltas}
              : {scores: {$push: {$each: deltas}}}
          ]
        }

      }
  newText = await context.prisma.$runCommandRaw(rawQuery)
```

How do I do this with Prisma-js?

**Comment by nurul3101** [comment]
Hey @geyang 👋 

You need to use the [push](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#push) method for achieving this.

Here's an example:

```javascript
const addScore = await prisma.students.update({
  where: {
    id: 1,
  },
  data: {
    scores: {
      push: 90,
    },
  },
})
```

Please note that you can only push one value at a time.

**Comment by geyang** [comment]
This only inserts one item. How do I insert multiple items?

**Reply by nurul3101** [reply]
You would need to loop over the values you want to insert and add it one by one.

**Reply by nurul3101** [reply]
Another solution is to use [set](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#set-1) operator if you know the final array values which you want to insert, it would override the existing values.

**Reply by geyang** [reply]
These are large lists with large extensions that gets sent every few seconds. I can do this with a `$push $each` combination in mongodb relatively easily.

**Comment by geyang** [comment]
That won’t do because these are large lists.

Would you kindly provide an example of using $runCommand escape hatch? I
looked for examples but could not find anything that works.

With mongodb standard bindings this is relatively easy.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
